src/cattr/gen.py

```diff
@@ -315,7 +369,19 @@ def make_dict_structure_fn(
             "    )",
         ]
 
-    total_lines = lines + post_lines + ["  return __cl(**res)"]
+    # At the end, we create the function header.
+    internal_arg_line = ", ".join([f"{i}={i}" for i in internal_arg_parts])
+    for k, v in internal_arg_parts.items():
+        globs[k] = v
+    lines.insert(0, f"def {fn_name}(o, _, *, {internal_arg_line}):")
```

Comment:
Hello @tinche, first of all, thanks for the amazing library!

I believe this line could be optimized even further by removing it and moving it to `total_lines` construction, since Python lists are not optimized for insertion at the beginning, so:

```suggestion
    total_lines = (
        [f"def {fn_name}(o, _, *, {internal_arg_line}):"]
        + lines
        + post_lines
        + ["  return __cl("]
        + [f"    {line}" for line in invocation_lines]
        + ["  )"]
    )
```

What do you think?

Reply:
It could, but it's not important for this code to be very optimized since it only runs once per class.

That said, I actually like the way it looks so I will apply this suggestion ;)

Reply:
Consider turning it into a single list expression using unpacking operators for the added lists

Reply:
@tinche Yeah, I figured, but still, it caught my eye, so I pointed it out. ;)

Thank you.